src/goto-programs/goto_contractor.cpp

```diff
@@ -225,28 +278,32 @@ goto_contractort::create_constraint_from_expr2t(expr2tc expr)
   case expr2t::expr_ids::lessthan_id:
     c = new ibex::NumConstraint(*vars, (*f)(*vars) < (*g)(*vars));
     break;
+  case expr2t::expr_ids::equality_id:
+    c = new ibex::NumConstraint(*vars, (*f)(*vars) = (*g)(*vars));
+    break;
   default:
     return nullptr;
   }
   return c;
 }
 
-bool goto_contractort::is_unsupported_operator(expr2tc expr)
+bool goto_contractort::is_unsupported_operator_in_constraint(
+  const expr2tc &expr)
 {
   expr2tc e = get_base_object(expr);
   return is_arith_expr(e) || is_constant_number(e) || is_symbol2t(e) ||
-         is_notequal2t(e) || is_equality2t(e) || is_not2t(e) ||
-         is_modulus2t(e) || is_or2t(e) || is_and2t(e);
+         is_notequal2t(e) || is_not2t(e) || is_modulus2t(e) || is_or2t(e) ||
+         is_and2t(e);
```

Comment:
I think you've implemented the support for `and` and `notequal`, right?

Reply:
Correct. I changed the function name. sorry for the confusion.